src/python/pants/core/goals/deploy.py

```diff
@@ -163,7 +164,7 @@ async def _invoke_process(
         return 0, tuple(results)
 
     logger.debug(f"Execute {process}")
-    res = await Effect(InteractiveProcessResult, InteractiveProcess, process)
+    res = await run_interactive_process(process)
```

Comment:
Looking at this change makes me a bit concerned along two dimensions:

- we need to remember to use the `run_interactive_process` helper version every time, or else we have bad behaviour
- if an existing external plugin uses the `Effect(...)`, they'll need to update or have the wrong behaviour

I have no idea how to mitigate either of these... but just an observation.

NB. Having a story for external plugins seems critical before releasing 2.22.0 stable.

Reply:
Yes, it's not awesome. But once we switch to call-by-name, the now "private" (underscore-led) name of the original rule will be a good warning not to call it directly.

Reply:
I could maybe add some kind of warning in the original rule that the helpers turn off, so you'll at least see something if you call it directly.

Reply:
Added a warning, although the way of doing so is a bit janky. But we don't need it forever.